packages/itwin/grouping-mapping-widget/src/components/Properties/GroupProperties/GroupPropertyUtils.ts

```diff
@@ -221,42 +231,52 @@ export const convertPresentationFields = (propertyFields: Field[]): PropertyMeta
             if (
               (nestedContentField.pathToPrimaryClass[0].relationshipInfo
                 .name === "BisCore:ElementOwnsUniqueAspect" ||
-                nestedContentField.pathToPrimaryClass[0].relationshipInfo
-                  .name === "BisCore:ElementOwnsMultiAspects")
+                  nestedContentField.pathToPrimaryClass[0].relationshipInfo
+                    .name === "BisCore:ElementOwnsMultiAspects")
             ) {
               const fullClassName = nestedContentField.contentClassInfo.name;
               const sourceSchema = fullClassName.split(":")[0];
               const sourceClassName = fullClassName.split(":")[1];
               const extractedNested = extractNested([], nestedContentField.nestedFields);
-              const aspectExtractedNested = extractedNested.map((ecProperty) => ({ ...ecProperty, sourceSchema, sourceClassName }));
-              return aspectExtractedNested;
+              extractedNested.forEach((ecProperty) => {
+                const propertyWithSchema = { ...ecProperty, sourceSchema, sourceClassName };
+                uniquePropertiesMap.set(propertyWithSchema.key, propertyWithSchema);
+              });
             }
             break;
           }
-          // Navigation properties
           case RelationshipMeaning.RelatedInstance: {
+            // Navigation properties
             if (
-              // Deal with a TypeDefinition
+            // Deal with a TypeDefinition
               nestedContentField.pathToPrimaryClass[0].relationshipInfo.name ===
-              "BisCore:GeometricElement3dHasTypeDefinition"
+                "BisCore:GeometricElement3dHasTypeDefinition"
             ) {
-              return extractNested(["TypeDefinition"], nestedContentField.nestedFields);
+              const extractedNested = extractNested(["TypeDefinition"], nestedContentField.nestedFields);
+              extractedNested.forEach((ecProperty) => {
+                uniquePropertiesMap.set(ecProperty.key, ecProperty);
+              });
             }
             break;
           }
           default: {
             // Some elements don't have a path to primary class or relationship meaning..
             // Most likely a simple struct property
             if (!nestedContentField.pathToPrimaryClass) {
-              return extractStruct(property);
+              const extractedStruct = extractStruct(property);
+              extractedStruct.forEach((ecProperty) => {
+                uniquePropertiesMap.set(ecProperty.key, ecProperty);
+              });
             }
           }
         }
       }
     }
-    return [];
   });
 
+  return Array.from(uniquePropertiesMap.values());
```

Comment:
Storing values in a map and then retrieving listing all of them might change the order, since Maps are not ordered (I think). Is there anything that ensures a consistent ordering of these properties?

Reply:
Maps remember insertion order. [https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map](url)